pkg/graveler/settings/manager.go

```diff
@@ -132,45 +132,50 @@ func (m *KVManager) Get(ctx context.Context, repositoryID graveler.RepositoryID,
 // The settingTemplate parameter is used to determine the type passed to the update function.
 func (m *KVManager) Update(ctx context.Context, repositoryID graveler.RepositoryID, key string, settingTemplate proto.Message, update updateFunc) error {
 	const (
-		maxIntervalSec = 2
-		maxElapsedSec  = 5
+		initialDurationMS = 100
+		maxDurationSec    = 5
+		maxSteps          = 10
+		factor            = 1.5
 	)
-	bo := backoff.NewExponentialBackOff()
-	bo.MaxInterval = maxIntervalSec * time.Second
-	bo.MaxElapsedTime = maxElapsedSec * time.Second
 
-	err := backoff.Retry(func() error {
+	err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
```

Comment:
I don't think we need another backoff package - we can simplify some of the code, like: 'bo.NextBackOff() == bo.Stop'.
But I don't see a reason to add a new package.

Reply:
It doesn't work as expected and at times fails